website/static/js/citationsFolderPickerViewModel.js

```diff
@@ -0,0 +1,487 @@
+        return userHasAuth && !nodeHasAuth && loaded;
+    });
+
+    /** Whether or not to show the full settings pane. */
+    self.showSettings = ko.computed(function() {
+        return self.nodeHasAuth();
+    });
+
+    /** Whether or not to show the Create Access Token button */
+    self.showTokenCreateButton = ko.computed(function() {
+        // Invoke the observables to ensure dependency tracking
+        var userHasAuth = self.userHasAuth();
+        var nodeHasAuth = self.nodeHasAuth();
+        var loaded = self.loadedSettings();
+        return !userHasAuth && !nodeHasAuth && loaded;
+    });
+
+    /** Computed functions for the linked and selected folders' display text.*/
+    self.folderName = ko.computed(function() {
+        var nodeHasAuth = self.nodeHasAuth();
+        var folder = self.folder();
+        return (nodeHasAuth && folder) ? folder.name : '';
+    });
+
+    self.selectedFolderName = ko.computed(function() {
+        var userIsOwner = self.userIsOwner();
+        var selected = self.selected();
+        return (userIsOwner && selected) ? selected.name : '';
+    });
+
+
+    // Initial fetch from server
+    self.fetchFromServer();
+};
+
+
+/** Change the flashed message. */
+CitationsFolderPickerViewModel.prototype.changeMessage = function(text, css, timeout) {
+    var self = this;
+
+    self.message(text);
+    var cssClass = css || 'text-info';
+    self.messageClass(cssClass);
+    if (timeout) {
+        // Reset message after timeout period
+        setTimeout(function() {
+            self.message('');
+            self.messageClass('text-info');
+        }, timeout);
+    }
+};
+
+/**
+ * Update the view model from data returned from the server.
+ */
+CitationsFolderPickerViewModel.prototype.updateFromData = function(data) {
+    var self = this;
+    self.ownerName(data.ownerName);
+    self.nodeHasAuth(data.nodeHasAuth);
+    self.userIsOwner(data.userIsOwner);
+    self.userHasAuth(data.userHasAuth);
+    self.userAccountId(data.userAccountId);
+    self.folder(data.folder || 'None');
+    self.urls(data.urls);
+    self.validCredentials(data.validCredentials);
+};
+
+CitationsFolderPickerViewModel.prototype.fetchFromServer = function() {
+    var self = this;
+    var request = $.ajax({
+        url: self.url,
+        type: 'GET',
+        dataType: 'json'
+    });
+
+    request.done(function(response) {
+        self.updateFromData(response);
+        self.loadedSettings(true);
+        if (!self.validCredentials()) {
+            if (self.userIsOwner()) {
+                self.changeMessage(self.messages.INVALID_CRED_OWNER(), 'text-warning');
+            } else {
+                self.changeMessage(self.messages.INVALID_CRED_NOT_OWNER(), 'text-warning');
+            }
+        }
+    });
+    request.fail(function(xhr, textStatus, error) {
+        self.changeMessage(self.messages.CANT_RETRIEVE_SETTINGS(), 'text-warning');
+        Raven.captureMessage('Could not GET Mendeley settings', {
```

Comment:
`"Mendeley"` will need to be changed to use the correct addon name. @samchrisinger